apps/03TicTacToe.js

```diff
@@ -22,25 +30,88 @@ function printBoard() {
   console.log('  ---------');
   console.log('2 ' + board[2].join(' | '));
 }
+function checkForWin() {
+  // All Win functions are inside this function.  
 
-function horizontalWin() {
-  // Your code here
+if (horizontalWin() || verticalWin() || diagonalWin()) {
+  return true;
+}
 }
+      
+
+
 
+function horizontalWin() {
+  // This chunk checks to see if there is a win on any of the rows. 
+  // Since this statement has a true/false if/else statement it doesn't need to be restated again. 
+  if (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn ||
+      board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn ||
+      board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn) {
+        return true;
+  }
+  else { 
+    return false;
+  }
+  }
+
+// This checks to see if the there is a win in the columns
 function verticalWin() {
-  // Your code here
+    if (board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn ||
+        board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn ||
+        board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn) {
+          return true;
+  }
+  else { 
+    return false;
+  }
+
 }
 
 function diagonalWin() {
-  // Your code here
-}
+      if (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn ||
+          board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn)  {
+            return true;
+  }
+  else { 
+    return false;
+  }
 
-function checkForWin() {
-  // Your code here
 }
 
+
+
+
 function ticTacToe(row, column) {
-  // Your code here
+  // Calls takes input from user and places it into row and column in the original board. 
+  board[row][column] = playerTurn;
+  // the next lines check for wins in the rows, columns and diagonals. 
+  if (checkForWin() === true) {
```

Comment:
If lines 89-103 only run when `checkForWin() === true`, you should place this logic inside that corresponding function.